### PR TITLE
Check for compile of misc.c and fix disable inline

### PR DESCRIPTION
**Pull request by JacobBarthelmeh**
Check for compile of misc.c (helpful for developers using IDE's)

fix disable inline

### CODE REVIEW

File: src/internal.c [hunk 1]
```diff
@@ -36,6 +36,7 @@
 #ifdef NO_INLINE
     #include <wolfssl/wolfcrypt/misc.h>
 #else
+    #define WOLFSSL_MISC_COMPILED
```
Comment (toddouska) [inline src/internal.c]: I believe this name is misleading.  While the file is being compiled it is NOT being compiled directly.  It is being included as if it were a header.

File: wolfcrypt/src/misc.c [hunk 2]
```diff
@@ -39,11 +39,18 @@
  */
 
 #ifdef NO_INLINE
+    #undef  WOLFSSL_MISC_COMPILED
```
Comment (toddouska) [inline wolfcrypt/src/misc.c]: I'm not sure it makes sense to define this when the user includes the file directly.  We're trying to distinguish the case of user explicit compile vs implicit header include compile, not make it more cloudy.

File: wolfcrypt/src/misc.c [hunk 3]
```diff
@@ -39,11 +39,18 @@
...
+    #define WOLFSSL_MISC_COMPILED
     #define STATIC
 #else
     #define STATIC static
 #endif
 
+/* Check for if compiling misc.c when not needed. FIPS build has source code
+   without preprocessor flag so the warning is not used in a FIPS build. */
+#if !defined(WOLFSSL_MISC_COMPILED) && !defined(HAVE_FIPS)
```
Comment (toddouska) [inline wolfcrypt/src/misc.c]: I think we'll no longer need to worry about FIPS if above guidance is followed.  The check more simply becomes 

! WOLFSSL_MISCC_INCLUDED && ! NO_INLINE

**Comment by JacobBarthelmeh** [comment]
Will update the PR to reflect the comments.
There may be an issue with source files in FIPS such as sha.c that include ctaocrypt/src/misc.c which is a redirect to wolfcrypt/src/misc.c. Since not changing sha.c in the FIPS boundary it would be including misc.c without the new macro. Maybe placing the macro in ctaocrypt/src/misc.c may be the way to go in this case, since it has the added benefit of not breaking reverse compatibility.

**Comment by toddouska** [comment]
Thanks, looks good.

**Pull request merged**